### PR TITLE
2pc: doc changes

### PR DESCRIPTION
**Pull request by sougou**
The design doc is updated, but the user guide still needs work.

### CODE REVIEW

File: doc/TwoPhaseCommitGuide.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+# Overview
+
+Vitess 2PC allows you to perform atomic distributed commits. The feature is implemented using traditional MySQL transactions, and hence inherits the same guarantees.
+
+The 2PC API itself is fairly simple: The Commit request accepts an additional `Atomic` flag. If it's set, then the 2PC mechanism is triggered. Without the flag, then a Commit is treated as best effort, which may result in partial commits if there are system failures during the operation. Additional extensions have been added to the API, which will be explained below.
+
+2PC commits are more expensive than best effort because the system has to save away the statements before starting the commit process, and also clean them up after a successful commit. The application can make the trade-off call based on when it believes a 2PC commit is warranted. For single-database transactions, there is no additional overhead; 2PC commits and best effort commits are the same cost.
+
+# gRPC changes
```
Comment (alainjobart) [inline doc/TwoPhaseCommitGuide.md]: gRPC is one transport, I'd say: API changes

File: doc/TwoPhaseCommitGuide.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+As part of the 2pc feature, the VTGate API has been extended to support three levels of Atomicity:
```
Comment (alainjobart) [inline doc/TwoPhaseCommitGuide.md]: This comment is about VTGate binary, not API.

File: doc/TwoPhaseCommitGuide.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+For Python, the `begin` function of the cursor now has an optional `single_db` flag, and the `commit` function has an optional `twopc` flag.
+
+## Java & PHP (TODO)
+
+# Enabling 2PC in Vitess
+
+## VTTablet
+
+* **twopc_enable**: This flag needs to be turned on.
+* **twopc_coordinator_address**: This should specify the address (or VIP) of the VTGate that VTTablte will use to resolve abandoned transactions.
```
Comment (alainjobart) [inline doc/TwoPhaseCommitGuide.md]: type on VTTablte

File: doc/TwoPhaseCommitGuide.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+* **twopc_abandon_age**: This is the time in seconds that specifies how long to wait before asking a VTGate to resolve an abaondoned transaction.
```
Comment (alainjobart) [inline doc/TwoPhaseCommitGuide.md]: abaondoned

File: doc/TwoPhaseCommitGuide.md (hunk 2 above)
Comment (sougou) [inline doc/TwoPhaseCommitGuide.md]: I think this is confusing overall. I'll prefix this with a Transaction Model section that explains the actual model, and then drill down on the actual details.

**Comment by sougou** [comment]
PTAL

**Pull request merged**